runner/src/bootstrap.rs

```diff
@@ -94,3 +97,25 @@ async fn random(n: usize, peers: &BTreeMap<usize, PeerInfo>) -> Result<()> {
     }
     Ok(())
 }
+
+#[tracing::instrument(skip(peers), fields(peers.len = peers.len()))]
+async fn sentinel(n: usize, peers: &BTreeMap<usize, PeerInfo>) -> Result<()> {
+    for peer_info in peers.values() {
+        // Connect to each peer to the first n peers
+        // Do not attempt to connect to more peers than exist.
+        for sentinel in 0..min(n, peers.len() - 1) {
+            debug!(%peer_info.index, %sentinel, "sentinel peer connection");
+            if let Some(sentinel_info) = peers.get(&sentinel) {
```

Comment:
Does it matter that we might call `connect_peers` with the same peer for both sides of the connection?

Reply:
It will pollute logs, as the API call will fail. I fixed this in #35 as I had to rewrite bits of it anyway.